WordPress/Classes/ViewRelated/Blog/My Site/MySiteSettings.swift

```diff
@@ -13,19 +16,19 @@ import Foundation
     }
 
     @objc var experimentAssignment: String {
-        let defaultSection: MySiteViewController.Section = .siteMenu
-        let rawValue = userDefaults.object(forKey: Constants.defaultSectionKey) as? Int ?? defaultSection.rawValue
-        return MySiteViewController.Section(rawValue: rawValue)?.analyticsDescription ?? "nonexistent"
+        if let rawValue = userDefaults.object(forKey: Constants.defaultSectionKey) as? Int,
+            let defaultSection = MySiteViewController.Section(rawValue: rawValue)?.analyticsDescription {
+            return defaultSection
+        }
+
+        return "nonexistent"
     }
 
     // MARK: - Init
 
-    init(userDefaults: UserDefaults = .standard) {
```

Comment:
Comment about this: by doing this injection `UserDefaults` was returning `nil` even though the value has been previously saved. I'm really not sure why this was happening.

Reply:
Weird!